submissions/f3d0t/memory_pair_game/app.js

```diff
@@ -0,0 +1,108 @@
+const CARDS_ARRAY = [
+	{
+		id: 1,
+		path: "img/1.png",
+	},
+	{
+		id: 2,
+		path: "img/2.png",
+	},
+	{
+		id: 3,
+		path: "img/3.png",
+	},
+	{
+		id: 4,
+		path: "img/4.png",
+	},
+	{
+		id: 5,
+		path: "img/5.png",
+	},
+	{
+		id: 6,
+		path: "img/6.png",
+	},
+	{
+		id: 7,
+		path: "img/7.png",
+	},
+	{
+		id: 8,
+		path: "img/8.png",
+	},
+];
+const CARDS_CONTAINER = document.getElementById("cards_container");
+const TURN_DELAY = 900;
+const openedNow = []; //will contain array of two cards(Dom objects), implemented to avoid bugs on checking
+let startTime = Date.now();
+
+const shuffleArray = (array) => array.sort(() => 0.5 - Math.random());
+
+const getCardHTML = (cardObject) => {
+	const card = document.createElement("div");
+	card.classList.add("card");
+	card.setAttribute("data-id", cardObject["id"]);
+	const cardFront = document.createElement("div");
+	cardFront.classList.add("card__front");
+	const cardBack = document.createElement("div");
+	cardBack.classList.add("card__back");
+	cardBack.style.backgroundImage = `url(${cardObject["path"]})`;
+	card.append(cardFront, cardBack);
+	return card;
+};
+
+const getAllCardsHTML = () => {
+	const containerFragment = document.createDocumentFragment();
+	const cardsShuffledArray = shuffleArray([...CARDS_ARRAY, ...CARDS_ARRAY]);
+	cardsShuffledArray.forEach((cardDataObject) => containerFragment.append(getCardHTML(cardDataObject)));
+	return containerFragment;
+};
+
+const checkPair = (twoCardsArray) => {
+	if (twoCardsArray[0].dataset.id == twoCardsArray[1].dataset.id) {
+		twoCardsArray.forEach((card) => {
+			setTimeout(() => {
+				card.classList.add("solved");
+			}, TURN_DELAY);
+		});
+	}
+	twoCardsArray.forEach((card) => {
+		setTimeout(() => {
+			card.classList.remove("flipped");
+		}, TURN_DELAY);
+	});
+};
+
+const checkIfWin = () => {
+    if (CARDS_CONTAINER.getElementsByClassName("solved").length == CARDS_CONTAINER.childNodes.length) {
```

Comment:
Do not rely on DOM itself, keep number of cards separately.

Reply:
Done, now it counts in separate value